File: 0791_Custom_Sort_String.cpp
class Solution {
public:
    string customSortString(string S, string T) {
        int iter = 0;
        char temp;
        for (int i = 0; i < S.size(); i++) {
            for (int j = iter; j < T.size(); j++) {
                if (S[i] == T[j]) {
                    temp = T[iter];
                    T[iter] = T[j];
                    T[j] = temp;
                    iter++;
                }
            }
        } 
        return T;
    }
};

File: 0791_Custom_Sort_String.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        counter = collections.Counter(s)
        res = []
        for k in order:
            if counter[k] > 0:
                res.extend([k] * counter[k])
                del counter[k]
        for k, v in counter.items():
            res.extend([k] * v)
        return ''.join(res)
